webapp/components/survey/NodeDefDetails/store/actions/useCancelNodeDefEdits.js

```diff
@@ -0,0 +1,19 @@
+import * as NodeDef from '@core/survey/nodeDef'
+
+import * as NodeDefActions from '@webapp/store/survey/nodeDefs/actions'
+
+import * as NodeDefState from '../state'
+
+export const useCancelNodeDefEdits = ({ nodeDefState }) => (history) => async (dispatch) => {
+  const nodeDef = NodeDefState.getNodeDef(nodeDefState)
+  const nodeDefOriginal = NodeDefState.getNodeDefOriginal(nodeDefState)
+
+  await dispatch({
+    type: NodeDefActions.nodeDefPropsUpdateCancel,
+    nodeDef,
+    nodeDefOriginal,
+    isNodeDefNew: NodeDef.isTemporary(nodeDef),
+  })
+
+  history.goBack()
+}
```

Comment:
do not mix redux actions with local store actions, plus do not export redux keys.
If an action updates only redux stores, than that action must be placed in redux store

Reply:
if you need to dispatch a redux actions within a local store action, you can use useDispatch()

Reply:
yes, I'll change it